tests/base/src/test/java/org/wildfly/security/auth/server/SaslExternalServerAuthenticationCallbackTest.java

```diff
@@ -0,0 +1,313 @@
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.Principal;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class SaslExternalServerAuthenticationCallbackTest {
+
+    static Path rootPath;
+    static SecurityDomain securityDomain;
+    static SetMechanismInformationSaslServerFactory factory;
+
+    @BeforeClass
+    public static void testSkipCertVerificationProp() throws Exception {
+        mockClientsCertificateEvidence();
+
+        // create a test user
+        FileSystemSecurityRealm securityRealm = new FileSystemSecurityRealm(getRootPath(), 3);
+        ModifiableRealmIdentity identity = securityRealm.getRealmIdentityForUpdate(new NamePrincipal("externalSaslUser"));
+        identity.create();
+        identity.dispose();
+        assertTrue(identity.exists());
+
+        // create security domain with CN principal decoder
+        securityDomain = SecurityDomain.builder().setDefaultRealmName("default").addRealm("default", securityRealm).build()
+                .setPermissionMapper((permissionMappable, roles) -> LoginPermission.getInstance())
+                .setPrincipalDecoder(new X500AttributePrincipalDecoder("2.5.4.3"))
+                .build();
+
+        // create external sasl server factory for tests
+        SaslServerFactory externalSaslServerFactory = new ExternalSaslServerFactory();
+        factory = new SetMechanismInformationSaslServerFactory(externalSaslServerFactory);
+        assertNotNull("SaslServerFactory not registered", factory);
+    }
+
+    @Test
+    public void testWithSkipCertificateVerificationProp() throws GeneralSecurityException, IOException, UnsupportedCallbackException {
+        ServerAuthenticationContext sac = securityDomain.createNewAuthenticationContext();
+        CallbackHandler callbackHandler = sac.createCallbackHandler();
+        SSLConnection sslConnection = SSLConnection.forSession(new DummySessionContainingPeerCertificates(), true);
+        callbackHandler.handle(new SSLCallback[]{new SSLCallback(sslConnection)});
+
+        SaslServer saslServerWithSkipCertificateProp = factory.createSaslServer("EXTERNAL", "test", "localhost", setProps("org.wildfly.security.sasl.skip-certificate-verification"), callbackHandler);
```

Comment:
Might be good to include a test for the case where the property is set to false.

Reply:
@fjuma Test that uses the property set to `false` is now added